[DataGrid] Add controlled behavior for the visible columns (#3554)
mui-org/material-ui-x: docs/src/pages/components/data-grid/columns/columns.md

{{"demo": "pages/components/data-grid/columns/ColumnTypesGrid.js", "bg": "inline"}}

## Column visibility

By default, all the columns are visible.

> The column's visibility can be switched through the grid interface in two ways:

Author: Done :+1:

- By opening the column menu and clicking the _Hide_ menu item.
- By clicking the _Columns_ menu and toggling the columns to show or hide.

You can prevent the user from hiding a column through the grid interface by setting the `hideable` in `GridColDef` to `false`.

In the following demo, the "username" column cannot be hidden.

{{"demo": "pages/components/data-grid/columns/VisibleColumnsBasicExample.js", "bg": "inline"}}

> **Note:** If a column is not registered in the `columnVisibilityModel`, it is considered visible.

Author: It also applied to the `initialState.columns.columnVisibilityModel`

> **Note:** If a column is not registered in the `columnVisibilityModel`, it is considered visible.

Author: I can change the wording to stop talking about the model

```suggestion
**Note:** If a column visibility is not defined, the column will be visible by default.
```

I would not like users only focusing on the `initialState` to miss that part.?

### Initialize the visible columns

To initialize the visible columns without controlling them, provide the model to the `initialState` prop.

```tsx
<DataGrid
  initialState={{
    columns: {

>       columnsVisibilityModel: { status: false },

Author: Done
